Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp

```diff
@@ -59,24 +59,51 @@ namespace AZ
             DestroySwapChain();
         }
 
-        const RHI::AttachmentId& WindowContext::GetSwapChainAttachmentId() const
-        {
-            return m_swapChain->GetAttachmentId();
+        const RHI::AttachmentId& WindowContext::GetSwapChainAttachmentId(SwapChainMode swapChainMode) const
+        {   
+            return GetSwapChain(swapChainMode)->GetAttachmentId();
         }
 
-        const RHI::Ptr<RHI::SwapChain>& WindowContext::GetSwapChain() const
+        const RHI::Ptr<RHI::SwapChain>& WindowContext::GetSwapChain(SwapChainMode swapChainMode) const
         {
-            return m_swapChain;
+            switch (swapChainMode)
+            {
+                case SwapChainMode::Default:
+                    return m_swapChain;
+                case SwapChainMode::XrLeft:
+                case SwapChainMode::XrRight:
```

Comment:
nit (repeated) - You can add a `[[fallthrough]];` statement between the left/right cases avoid warnings.  Although I don't know if any of our supported compilers have fall-through warnings turned on currently.